manifests/install.pp

```diff
@@ -118,6 +118,72 @@
           ensure => $docker::ensure,
           name   => $dockerpackage,
         }
+      }
+    }
+      
+    if $docker::docker_compose {
```

Comment:
Mmm, so I would be tempted to pull this out into a separate class, or even a separate module to begin with.

* Having the docker package explicitly manage the python-dev tools isn't a good idea, it prevents them being managed elsewhere - for instance in the [python module](https://forge.puppetlabs.com/stankevich/python).
* Ditto the pip dependencies
* EPEL is not optional based on a passed param

I don't think for the moment this should be included. I'm happy for installing docker-compose to be an exercise left up to the user, maybe with an example of how to do that in puppet I think.